core/app/models/spree/order.rb

```diff
@@ -596,8 +596,9 @@ def add_store_credit_payments
 
       if matching_store_credits.any?
         payment_method = Spree::PaymentMethod::StoreCredit.first
+        sorter = Spree::Config.order_credit_sorter_class.new(self)
 
-        matching_store_credits.order_by_priority.each do |credit|
+        sorter.in_sorted_order(matching_store_credits).each do |credit|
```

Comment:
What about just using `call` here? It is a sorter, and sorting is the only thing it is expected to do, so I guess `sorter.call` is enough semantic. 

Also, if you think my previous suggestion makes sense, this could be without any argument in the core, you could change the signature in your own implementation, passing the order. WDYT?